Feature/set protocol docu (#126)
eclipse/kuksa.val: doc/protocol/set.md

## set: Write data

> KUKSA.val supports basic set requests according to VISS  validated against the following schema.

Reviewer: ```suggestion
KUKSA.val supports basic set requests according to VISS validated against the following schema.
```

```json

> {

Reviewer: better reference to https://github.com/eclipse/kuksa.val/blob/master/include/VSSRequestJsonSchema.hpp#L48-L71 instead of copy the code?

    "$schema": "http://json-schema.org/draft-04/schema#",
    "title": "Set Request",
    "description": "Enables the client to set one or more values once.",
    "type": "object",
    "required": ["action", "path", "value", "requestId"],
    "properties": {
        "action": {
            "enum": [ "set" ],
            "description": "The identifier for the set request"
        },
        "path": {
            "$ref": "viss#/definitions/path"
        },
        "value": {
            "$ref": "viss#/definitions/value"
        },
        "requestId": {
            "$ref": "viss#/definitions/requestId"
        }
    }
}
```

The following shows some valid and invalid sample requests. Unless otherwise noticed, we assume you are authorised to read the paths used in the examples.

### VSS1-Path Example Request

> KUKSA.val supports VISS1 (dot-seperated) and VISS2  (slash-seperated) paths. VISS1 example

Reviewer: ```suggestion
KUKSA.val supports VISS1 (dot-seperated) and VISS2 (slash-seperated) paths. VISS1 example:
```

```json
{
  "action": "set",
  "path": "Vehicle/VehicleIdentification/VIN",
  "value": "deadbeef",
  "requestId": "4544fd5f-249e-4da6-8576-248b42504394"
}
```

### Response

```json
{
    "action": "set", 
    "error": {
        "message": "Can not set Vehicle/VehicleIdentification/VIN. Only sensor or actor leaves can be set.", 
        "number": 403, 
        "reason": "Forbidden"
    }, 
    "requestId": "4544fd5f-249e-4da6-8576-248b42504394", 
    "timestamp": 1615833005144
}

```

### Non-existing path
As expected, trying to set a non-existent path returns a 404

```json
{
  "action": "set",
  "path": "Vehicle/FluxCapacitor/Charge",
  "value": "1.21",
  "requestId": "882a6bdd-ff74-433a-a7c4-c2ce2c7f5c07"
}
```

### Response

```json
{
    "action": "set", 
    "error": {
        "message": "I can not find Vehicle/FluxCapacitor/Charge in my db", 
        "number": 404, 
        "reason": "Path not found"
    }, 
    "requestId": "882a6bdd-ff74-433a-a7c4-c2ce2c7f5c07", 
    "timestamp": 1615833326191
}
```

### Authorization Checks first

> Please not, that authrization is checked *first*, i.e. in the previous example, if we do not have a permission covering the requested path, no matter it exists or not, we get a 403

Reviewer: ```suggestion
Please note, that authorization is checked *first*, i.e. in the previous example, if we do not have a permission covering the requested path, no matter it exists or not, we get a 403
```

```json
{
  "action": "set",
  "path": "Vehicle/FluxCapacitor/Charge",
  "value": "1.21",
  "requestId": "458a269f-ca42-4175-b88c-5ff8bea7855e"
}
```

### Response

```json
{
    "action": "set", 
    "error": {

>         "message": "No write  access to Vehicle/FluxCapacitor/Charge", 

Reviewer: ```suggestion
        "message": "No write access to Vehicle/FluxCapacitor/Charge", 
```

Reviewer: maybe also change in code

Author: maybe, but no in this PR. Also, will not change the Readme for now (that would be cheating)
